Update documentation to remove formation and add css-library

powellkerry opened the pull request:
Update all references and links to formation with the correct information and links to css-library. This is needed due to formation being deprecated and no longer being used in vets-website.

Closes [3583](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3583)

Note, this does not update code references to formation. That work is planned in [this ticket](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3621)

powellkerry commented on README.md:
```diff
@@ -57,23 +57,23 @@ To add content, you will need to look into `/src` directory. This will be the so
 
-## Testing updates to the Formation codebase on this site
+## Testing updates to the CSS-Library codebase on this site
 
```
I just updated this copy and I'm not sure if there is a better way to give these instructions or if we need to update them.

micahchiang replied:
@powellkerry did you happen to test the steps below to see if they work for css library? I'm specifically interested in learning if that local version thing on line 74 works, since that seems to be the key piece for connecting the two repos for local testing.

powellkerry replied:
@micahchiang yes the instructions work, but it's not as easy as they imply. I had to remove the node_modules directory and run yarn in vets-design-system-documentation in order for any changes to show. I think updating the instructions to use verdaccio might be better?

powellkerry commented on src/_foundation/spacing-units.md:
```diff
@@ -46,7 +46,13 @@ The VA follows the USWDS spacing unit tokens and then adds additional semantic t
 
-In order to access the spacing tokens in your project, you will need to import the [base files](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/tree/master/packages/formation/sass/base) into your project. [Here is how this site is doing that](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/blob/main/src/assets/stylesheets/applicat ...
+In order to access the spacing tokens in your project, you will need to import the [base files](https://github.com/department-of-veterans-affairs/component-library/tree/main/packages/css-library/src/stylesheets) into your project. 
+<!-- 
```
I removed this link to look at an example because the code shows code references to formation. I will leave a comment on the ticket for updating the code references to re-add this link.

micahchiang replied:
This piece about spacing tokens is a little confusing because the paragraph above references a units function :thinking: 

Can spacing tokens be accessed directly be importing https://github.com/department-of-veterans-affairs/component-library/blob/main/packages/css-library/dist/tokens/css/variables.css?

powellkerry replied:
I updated this. I think the spacing tokens can be used by importing that variables file.

powellkerry commented on src/_about/developers/install.md:
````diff
@@ -11,24 +11,21 @@ anchors:
+If you prefer to change the location of the `fonts/` and `img/` directories relative to `core.css`, set the following variables in your project:
 
 ```
-$formation-asset-path: '../assets';
````
I couldn't find a `asset-path` variable in css-library. Should this be updated or removed?

micahchiang replied:
I think for now, we should mark the existing stuff as deprecated, with a message that new guidance is coming soon. We should probably conduct our own spike or experiment into what is needed to get css-library working properly across a number of different projects. I can write a follow up ticket for this.

jamigibbs commented on src/_about/developers/install.md:
```diff
@@ -1,9 +1,10 @@
-intro-text: 
+intro-text: "New guidance in coming soon. With the deprecation of Formation and the adoption off CSS Library the design system team is in the process of creating and providing guidance on using CSS Library across different projects."
 anchors:
   - anchor: Install CSS library into your project
```
Capitalize the "L" in CSS library.

jamigibbs reviewed (commented):
Updates are looking good to me but do you mind checking the developer > install page again? https://dev-design.va.gov/3650/about/developers/install#install-css-library-into-your-project

Do we want to keep all of those references to Formation still even if we are planning to re-write the instructions? The header of the section is talking about CSS Library so it's a mix of libraries and confusing.

Do we have a ticket yet for discovering how to install CSS Library in different scenarios outside of vets-website? (Code Pen, Code Sandbox, etc)?

powellkerry commented:
@jamigibbs I updated install.md to say formation instead of css-library. I had just reverted my changes to that file and didn't notice that it mentioned css library. Evidently when this was created, they used to refer to formation as the css library...

I also created a ticket to research and rewrite this file: https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3663

jamigibbs reviewed (approved):
> Evidently when this was created, they used to refer to formation as the css library

Oh dear. 🫣
